### ISSUE TITLE
replace object-hash with JSON.stringify

### ISSUE THREAD

**Issue by gajus** [andreiduca/use-async-resource#49]
There is no obvious reason for using `object-hash` here and it adds 32kb to the bundle size.

https://bundlephobia.com/result?p=use-async-resource@2.2.1

**Comment by andreiduca** [comment]
Agreed, `object-hash` is large enough, but `JSON.stringify` doesn't preserve object equivalence: for identical objects with different order of their keys, it results in different values.

Another thing to keep note of: very large objects will result in very large keys, whereas a hash function will have the same length regardless of the object sizes.

There are alternatives, smaller and (maybe) faster. Too many actually. But I have not done much research:
- https://www.npmjs.com/package/fast-json-stable-stringify
- https://www.npmjs.com/package/json-stable-stringify
- https://www.npmjs.com/package/fast-stable-stringify
- https://www.npmjs.com/package/faster-stable-stringify

**Comment by miguelski** [comment]
also scope hoist in parcel (not sure about webpack)  doesn't work because of object-hash

**Comment by andreiduca** [comment]
I will look into alternatives. Thank you for bringing it up. 🙏🏻

**Comment by gajus** [comment]
Any chance of this getting addressed? It is currently being flagged as the biggest dead weight in our app.
